templates/beego/files/controllers/services.go

```diff
@@ -0,0 +1,139 @@
+package controllers
+
+import (
+        "fmt"
+        "database/sql"
+        _ "github.com/go-sql-driver/mysql"
+        sqldsn "github.com/platformsh/config-reader-go/v2/sqldsn"
+)
+
+type MySQLTestController struct {
+        MainController
+}
+
+func (c *MySQLTestController) Get() {
+
+  // Accessing the database relationship Credentials struct
+  credentials, err := c.Config.Credentials("database")
+  if err != nil {
+    panic(err)
+  }
+
+  // Using the sqldsn formatted credentials package
+  formatted, err := sqldsn.FormattedCredentials(credentials)
+  if err != nil {
+    panic(err)
+  }
+
+  db, err := sql.Open("mysql", formatted)
+  if err != nil {
+    panic(err)
+  }
+
+  // Force MySQL into modern mode.
+  db.Exec("SET NAMES=utf8")
+  db.Exec("SET sql_mode = 'ANSI,STRICT_TRANS_TABLES,STRICT_ALL_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,ONLY_FULL_GROUP_BY'")
+
+  _, err = db.Exec("DROP TABLE IF EXISTS userinfo")
+  if err != nil {
+    panic(err)
+  }
+
+  _, err = db.Exec(`CREATE TABLE userinfo (
+      uid INT(10) NOT NULL AUTO_INCREMENT,
+      username VARCHAR(64) NULL DEFAULT NULL,
+      departname VARCHAR(128) NULL DEFAULT NULL,
+      created DATE NULL DEFAULT NULL,
+      PRIMARY KEY (uid)
+      ) DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;`)
+  if err != nil {
+		panic(err)
+	}
+  // insert
+  stmt, err := db.Prepare("INSERT userinfo SET username=?,departname=?,created=?")
+  if err != nil {
+    panic(err)
+  }
+
+  res, err := stmt.Exec("platform", "Deploy Friday", "2019-06-17")
+  if err != nil {
+    panic(err)
+  }
+
+  id, err := res.LastInsertId()
+  if err != nil {
+    panic(err)
+  }
+
+  fmt.Println(id)
+  // update
+  stmt, err = db.Prepare("update userinfo set username=? where uid=?")
+  if err != nil {
+    panic(err)
+  }
+
+  res, err = stmt.Exec("goPlatformsh", id)
+  if err != nil {
+    panic(err)
+  }
+
+  affect, err := res.RowsAffected()
+  if err != nil {
+    panic(err)
+  }
+
+  fmt.Println(affect)
+
+  // query
+  rows, err := db.Query("SELECT * FROM userinfo")
+  if err != nil {
+    panic(err)
+  }
+
+  var uid int
+  var username string
+  var department string
+  var created string
+  for rows.Next() {
+    err = rows.Scan(&uid, &username, &department, &created)
+    if err != nil {
+      panic(err)
+    }
+  }
+
+  // delete
+  stmt, err = db.Prepare("delete from userinfo where uid=?")
+  if err != nil {
+    panic(err)
+  }
+
+  res, err = stmt.Exec(id)
+  if err != nil {
+    panic(err)
+  }
+
+  affect, err = res.RowsAffected()
+  if err != nil {
+    panic(err)
+  }
+
+  db.Close()
```

Comment:
I realized when poking at the language-examples version, this line should probably be moved to `defer db.Close()` right after the Open() call's error handling.  That's the more Go-ish way of making sure things get closed.  (Not a merge blocker, but something we should probably change across all the examples.)

Reply:
Like this?

```
  db, err := sql.Open("mysql", formatted)
  if err != nil {
    panic(err)
  }
defer db.Close()
```